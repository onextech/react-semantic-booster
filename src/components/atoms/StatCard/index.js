import React from 'react';
import PropTypes from 'prop-types';
import { Header } from 'semantic-ui-react';
import styled from 'styled-components';
import Card from '../Card';


const StyledCard = styled(Card)`
  &.ui.card {
    font-size: 200%;
  }
  
  &.ui.card > .content,
  &.ui.cards > .card > .content {
    padding: 3em 1em;
  }
  
  &.ui.card>.content>.header,
  &.ui.cards>.card>.content>.header {
    ${({ gradient }) => gradient && `
      background: -webkit-linear-gradient(left,${gradient.from},${gradient.to});
      -webkit-background-clip: text;
      -webkit-text-fill-color: transparent;
    `}
  }
`;

const StatNumber = styled(Header)`
  &.ui.header {
    margin-bottom: 0;
    font-weight: normal;
  }
`;

const StatHeader = styled(Header)`
  &.ui.header {
    margin-top: 0;
    font-weight: normal;
  }
`;

const StatDesc = styled(Card.Description)`
  font-size: 70%;
  line-height: 1.3;
`;

const StatCard = ({ card, content, headers }) => (
  <StyledCard fluid gradient={headers && headers.gradient} {...card}>
    <Card.Content textAlign="center">
      <StatNumber as="h5" size="huge" {...headers}>{content.number}</StatNumber>
      <StatHeader as="h5" size="medium" {...headers}>{content.header}</StatHeader>
      <StatDesc>{content.description}</StatDesc>
    </Card.Content>
  </StyledCard>
);

StatCard.propTypes = {
  card: PropTypes.object, // semantic-ui card props
  content: PropTypes.object,
  headers: PropTypes.object,
};

export default StatCard;
