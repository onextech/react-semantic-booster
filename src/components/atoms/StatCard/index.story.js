import React from 'react';
import { storiesOf } from '@storybook/react';
import { Grid } from 'semantic-ui-react';
import StatCard from '.';


const style1 = {
  card: {
    raised: true,
    basic: true,
  },
  headers: {
    color: 'blue',
    gradient: {
      from: '#1f40e6',
      to: '#37dee7',
    },
  },
};

const style2 = {
  card: {
    raised: true,
    stretched: true,
  },
  headers: {
    color: 'blue',
    gradient: {
      from: '#1f40e6',
      to: '#37dee7',
    },
  },
};

export const style0Example = [
  {
    content: {
      number: '60',
      header: 'world-class specialists',
      description: 'hired and onboarded',
    },
  },
  {
    content: {
      number: '8',
      header: 'agreements signed',
      description: 'with agencies, startups, and manufacturers',
    },
  },
  {
    content: {
      number: '1',
      header: 'fully-operational business',
      description: 'ready to capture the IoT opportunity',
    },
  },
];
export const style1Example = [
  {
    ...style1,
    content: {
      number: '60',
      header: 'world-class specialists',
      description: 'hired and onboarded',
    },
  },
  {
    ...style1,
    content: {
      number: '8',
      header: 'agreements signed',
      description: 'with agencies, startups, and manufacturers',
    },
  },
  {
    ...style1,
    content: {
      number: '1',
      header: 'fully-operational business',
      description: 'ready to capture the IoT opportunity',
    },
  },
];
export const style2Example = [
  {
    ...style2,
    content: {
      number: '60',
      header: 'world-class specialists',
      description: 'hired and onboarded',
    },
  },
  {
    ...style2,
    content: {
      number: '8',
      header: 'agreements signed',
      description: 'with agencies, startups, and manufacturers',
    },
  },
  {
    ...style2,
    content: {
      number: '1',
      header: 'fully-operational business',
      description: 'ready to capture the IoT opportunity',
    },
  },
];

storiesOf('StatCard', module)
  .add('Default', () => (
    <div>
      <Grid container stackable>
        <Grid.Row columns="equal">
          {style0Example.map(stat => (
            <Grid.Column key={stat.content.number}>
              <StatCard {...stat} />
            </Grid.Column>
          ))}
        </Grid.Row>
      </Grid>
      <Grid container stackable divided>
        <Grid.Row columns="equal">
          {style1Example.map(stat => (
            <Grid.Column key={stat.content.number}>
              <StatCard {...stat} />
            </Grid.Column>
          ))}
        </Grid.Row>
      </Grid>
      <Grid container stackable>
        <Grid.Row columns="equal">
          {style2Example.map(stat => (
            <Grid.Column key={stat.content.number}>
              <StatCard {...stat} />
            </Grid.Column>
          ))}
        </Grid.Row>
      </Grid>
    </div>
  ));
