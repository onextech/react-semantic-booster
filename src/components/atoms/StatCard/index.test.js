import React from 'react';
import renderer from 'react-test-renderer';
import StatCard from '.';
import { style0Example } from './index.story';


describe('<StatCard>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<StatCard {...style0Example[0]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
