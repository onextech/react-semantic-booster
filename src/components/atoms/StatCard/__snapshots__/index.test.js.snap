// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatCard> renders correctly 1`] = `
<div
  className="ui fluid card    StatCard__StyledCard-fUXdfT ndeBq Card__StyledCard-hKtLXx isCsom"
  gradient={undefined}
  href={undefined}
  onClick={[Function]}
>
  <div
    className="center aligned content"
  >
    <h5
      className="ui huge header StatCard__StatNumber-cUVTlO blSQes"
    >
      60
    </h5>
    <h5
      className="ui medium header StatCard__StatHeader-hDZpFU MNTII"
    >
      world-class specialists
    </h5>
    <div
      className="description StatCard__StatDesc-bvgQEi cWcpLY"
    >
      hired and onboarded
    </div>
  </div>
</div>
`;
